kedro/pipeline/node.py

```diff
@@ -511,6 +512,83 @@ def _process_inputs_for_bind(inputs: None | str | list[str] | dict[str, str]):
             kwargs = inputs
         return args, kwargs
 
+    def to_dict(self) -> NodeAsDict:
+        """Create a dictionary representation of the node suitable for dumping to JSON.
+
+        To ensure this representation can be loaded back in the same environment using
+        :py:meth:`kedro.pipeline.node.Node.from_dict`, the following constraints apply
+        to the node function:
+
+        - It cannot be a lambda function.
+        - It cannot be an inner function.
+        - It cannot be defined in ``__main__`` module.
+        """
+        err_msg = f"Cannot convert node '{self}' to dict."
+        qualname = self._func.__qualname__
+        if qualname == "<lambda>":
+            raise ValueError(f"{err_msg} Its function is a lambda function.")
+
+        if "<inner>" in qualname:
+            raise ValueError(f"{err_msg} Its function is an inner function.")
+
+        module = self._func.__module__
+        if module == "__main__":
+            raise ValueError(f"{err_msg} Its function's module is __main__.")
+
+        return {
+            "func": f"{self._func.__module__}.{self._func.__qualname__}",
+            # using the original values of the instance variables in order not to
```

Comment:
I'd like to reach out to Nichita and Ivan on whether we can steal anything from Alloy to detect requirements? The hardest piece in this is `pip install sklearn` vs `import scipy`